mcp/pointing/tests/test_pointing.c

```diff
@@ -2,24 +2,449 @@
 #include <stdarg.h>
 #include <stddef.h>
 #include <cmocka.h>
+#include <float.h>
+
 #include "pointing.c"
 
 #include "mcp_mock_decl.c"
 
 
+// ============================================================================
+// Setup/teardown functions (text fixtures)
+// ============================================================================
+/**
+ * @brief Set up the structs for El Solution tests
+ */
+static int SetupElSolution(void **state)
+{
+    *state = calloc(1, sizeof(struct ElSolutionStruct));
+    const struct ElSolutionStruct ElSol = {
+        .angle = 45.0,
+        .variance = 0.5,
+        .samp_weight = 1.0,
+        .sys_var = 0.5,
+        .trim = 0.0,
+        .last_input = 45.0,
+        .gy_int = 0.0,
+        .offset_gy = 1.0,
+        .FC = 1.0,
+        .n_solutions = 0,
+        .since_last = 0,
+        .fs = (struct FirStruct *) balloc(fatal, sizeof(struct FirStruct)),
+        .new_offset_ifel_gy = 0.0,
+        .int_ifel = 0.0,
+        .prev_sol_el = 45.0
+    };
+    init_fir(ElSol.fs, FIR_LENGTH, 0, 0);
+    memcpy(*state, &ElSol, sizeof(struct ElSolutionStruct));
+    return 0;
+}
+
+/**
+ * @brief Tear down the structs for El Solution tests
+ */
+static int TearDownElSolution(void **state)
+{
+    free(*state);
+    return 0;
+}
+
+/**
+ * @brief Set up the structs for Az Solution tests
+ */
+static int SetupAzSolution(void **state)
+{
+    *state = calloc(1, sizeof(struct AzSolutionStruct));
+
+    const struct AzSolutionStruct AzSol = {
+        .angle = 45.0,    // solution's current angle
+        .variance = 0.5, // solution's current sample variance
+        .samp_weight = 1.0, // sample weight per sample
+        .sys_var = 0.5,  // sytematic variance - can't do better than this
+        .trim = 0.0, // externally set trim to solution
+        .last_input = 45.0, // last good data point
+        .ifroll_gy_int = 1.0, // integral of the gyro since the last solution
+        .ifyaw_gy_int = 1.0,// integral of the gyro since the last solution
+        .offset_ifroll_gy =  1.0, // offset associated with solution
+        .offset_ifyaw_gy = 1.0,
+        .FC = 1.0, // filter constant
+        .n_solutions = 0, // number of angle inputs
```

Comment:
Just a nitpick, the spacing between the code and the inline comments is rather inconsistent. Please, fix.

Reply:
Fixes pushed